src/r2mm/manager/PackageDexieStore.ts

```diff
@@ -123,6 +126,40 @@ export async function getPackagesByNames(community: string, packageNames: string
     return packages.map(ThunderstoreMod.parseFromThunderstoreData);
 }
 
+/**
+ * @param community Community which package listings should be used in the lookup.
+ * @param dependencies Lookup targets as Thunderstore dependency strings.
+ * @param useLatestVersion Ignore the version number in dependencyString and return the latest known version.
+ * @returns ThunderstoreCombo[], silently omitting unknown packages and versions.
+ */
+export async function getCombosByDependencyStrings(
+    community: string,
+    dependencyStrings: string[],
+    useLatestVersion=false
+): Promise<ThunderstoreCombo[]> {
+    const split = dependencyStrings.map(splitToNameAndVersion);
+    const keys = split.map((d): [string, string] => [community, d[0]]);
+
+    // Dexie's anyOfIgnoreCase doesn't support compound indexes.
+    const packages = await db.packages.where('[community+full_name]').anyOf(keys).toArray();
+    const versionMap = Object.fromEntries(split);
+
+    return packages.map((rawPackage) => {
+        const rawVersion = useLatestVersion
+            ? rawPackage.versions[0]
+            : rawPackage.versions.find((v) => v.version_number === versionMap[rawPackage.full_name])
+
+        if (!rawVersion) {
+            return undefined;
+        }
+
+        const combo = new ThunderstoreCombo();
+        combo.setMod(ThunderstoreMod.parseFromThunderstoreData(rawPackage));
+        combo.setVersion(ThunderstoreVersion.parseFromThunderstoreData(rawVersion));
+        return combo;
+    }).filter((c): c is ThunderstoreCombo => c !== undefined);
```

Comment:
Two points on this:
1. Not a huge fan of this syntax. Do we need to use `(c): c is ThunderstoreCombo` instead of `(c: ThunderstoreCombo?) => c !== undefined`
2. Do we need to cast it? `c => c !== undefined)` is arguably more concise and gets the point across clearer. There may be an ESLint rule blocking this, so I'm happy for that to be the answer.

If neither of these strike you too much, I'm also not opposed to a util method to simply change this to something like: `.filter(ObjectUtils.onlyDefinedValues)`

Reply:
It's not about ESLint. Does the repo even use ESLint? 

Anyway, semantically speaking, I think the user-defined type guard is considered "type narrowing", and not brute force "type casting". AFAIK using "x is y" type guard functions is perfectly fine and TypeScriptic way of doing things like this.

In this particular case it's optional (due to simplicity of the case), and can be left out. At the same time the code base uses things like `private packageUrl: string = '';` where the `string` part is fully optional and not needed by TypeScript. But someone has considered it clearer to include these extra type descriptions only for the readers benefit. I don't think using the type guard here is any different. And when discussion is based on personal preferences, the proper solution would be to introduce linting to automate forcing the conventions instead of discussing them in code reviews. But that's outside the scope.

Reply:
> It's not about ESLint. Does the repo even use ESLint?

Potentially not. It did at one point and I think my configuration still has remnants of it.

> Anyway, semantically speaking, I think the user-defined type guard is considered "type narrowing", and not brute force "type casting". AFAIK using "x is y" type guard functions is perfectly fine and TypeScriptic way of doing things like this.

I'm fine with type guards where we actually need them, I don't think we've really used them much on the project and they don't appear to really add any benefit here as we don't care about the type of the filter. Supposedly we can do
```ts
.filter<ThunderstoreCombo>(c => c !== undefined)
``` 
which might read a bit nicer?

> At the same time the code base uses things like private packageUrl: string = ''; where the string part is fully optional and not needed by TypeScript

That looks like remnants of ESLint, granted I do prefer type declarations on properties themselves.

Reply:
> `.filter<ThunderstoreCombo>(c => c !== undefined)`

Fails for me on dev server build: `TS2345: Argument of type '(c: ThunderstoreCombo | undefined) => boolean' is not assignable to parameter of type '(value: ThunderstoreCombo | undefined, index: number, array: (ThunderstoreCombo | undefined)[]) => value is ThunderstoreCombo'.`

Did not change.